File: test_db.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore

# Use a service account
cred = credentials.Certificate('/home/daniel/programming/CBT-Dan/serviceAccountKey.json')
firebase_admin.initialize_app(cred)

db = firestore.client()

users_ref = db.collection('users')
docs = users_ref.stream()

teacher_id = None
for doc in docs:
    data = doc.to_dict()
    if data.get('email') == 'testteacher@test.com':
        print(f"Found teacher: {doc.id} => {data}")
        teacher_id = doc.id
        # Approve the teacher
        doc.reference.update({'isAuthorized': True})

if teacher_id:
    # Find a class to assign to this teacher
    classes_ref = db.collection('classes')
    class_docs = classes_ref.stream()
    for c_doc in class_docs:
        print(f"Assigning class {c_doc.id} to teacher {teacher_id}")
        c_doc.reference.update({'teacherId': teacher_id})
        break # Just one class is enough

File: test_db.js
import { initializeApp } from "firebase/app";
import { getFirestore, collection, getDocs } from "firebase/firestore";

const firebaseConfig = {
    apiKey: "dummy",
    authDomain: "cognita-44dbd.firebaseapp.com",
    projectId: "cognita-44dbd",
    storageBucket: "cognita-44dbd.firebasestorage.app",
    messagingSenderId: "dummy",
    appId: "dummy"
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

async function test() {
    const users = await getDocs(collection(db, "users"));
    users.forEach(doc => {
        if(doc.data().email === 'testteacher@test.com') {
            console.log("Found teacher:", doc.id, doc.data());
        }
    });
}
test();
